pyteal/ast/subroutine.py

```diff
@@ -66,10 +78,16 @@ def __init__(
         self.declaration: Optional["SubroutineDeclaration"] = None
         self.__name = self.implementation.__name__ if nameStr is None else nameStr
 
-    def getDeclaration(self) -> "SubroutineDeclaration":
+    def isRefArg(self, arg: str) -> bool:
+        anns = self.implementation.__annotations__
+        return arg in anns and anns[arg] is ScratchVar
+
+    def getDeclaration(
+        self, callback: "SubroutineCall" = None
+    ) -> "SubroutineDeclaration":
         if self.declaration is None:
             # lazy evaluate subroutine
-            self.declaration = evaluateSubroutine(self)
+            self.declaration = evaluateSubroutine(self, callback=callback)
```

Comment:
Code smell: lazy evaluation is no longer valid because a parameter is being introduced. In this case, I believe we're ok (when you follow the branches), but it's preferred to either:
* get rid of the laziness here and take a performance hit
* or split this into a new method that expects the parameter and isn't lazy

Reply:
**JUST PLAIN WRONG** ...gonna fix